### ISSUE TITLE
My Applications dashboard page (read only)

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/gc-digital-talent#3415]
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description
Create the My Applications dashboard page.
Should be rendered at `/user/{:userId}/applications`.

QUESTION: how should applications be sorted?
ANSWER: Sort by status only. Archived items hidden unless "show all" is clicked.

The details and style of the card depend on the status of the application. There are many new statuses and ALL statuses need specific copy. @Jerryescandon to add all that to the figma prototype. DCMO is still confirming statuses as well.

(leaving this list, but might not be relevant anymore)
- Draft
- Draft - expired (if pool closes... but Gray is not sure this should be a status. Maybe it should be programmatically determined)
- Application Submitted - New
- Under review (this means not a draft, but not yet accepted or declined)
- Accepted (meaning an admin has accepted the candidate into the pool)
- Not accepted (meaning an admin has rejected the candidate from the pool)
- Expired (meaning they were accepted, but expiry date has passed)
- TODO: what about drafts where the submit date has passed?

In the future an application may also be Archived, but ignore that for now.

`Continue my application` (for drafts) and `See advertisement` (for all cards) links should be functional.

`Delete` and `Archive` buttons should be built, but disabled and non-functional for this ticket.

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>

## Acceptance Criteria
- [x] Should be rendered at `/user/{:userId}/applications`
- [x] Given a user id, should fetch all PoolCandidates belonging to the user (using the poolCandidates field of the `user` query) and display them
- [x] Going to the page should force a log in, if not yet authenticated
- [x] `Continue my application` (for drafts) and `See advertisement` (for all cards) links should be functional.
- [x] Details and appearance of application card should depend on status (see prototype)
- [x] Delete and Archive buttons, and `See all applications` button should be built but left disabled and without functionality

   [image]: https://user-images.githubusercontent.com/4313717/179854868-5f59fbcd-9e0a-4ced-92e6-2d134bc144d9.png
   [image-link]: https://www.figma.com/proto/N9NoPunkqaynjsi89Wrbh0/Direct-Intake---Poster-and-application?node-id=228%3A4886&starting-point-node-id=226%3A4567
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

Out of scope 
- [ ] Applications sorting logic  #3807

**Comment by tristan-orourke** [comment]
Blocked by #3332 because we need to have decided how "draft" status is implemented.

**Comment by Jerryescandon** [comment]
| DCMO Status | Public facing label | Has snapshot? | Can be archived? | Can be deleted?|
| --------------- | -------------------- | ---------------- | ------------------- | ----------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |

**Comment by Jerryescandon** [comment]
Updated the table, removed some statuses

**Comment by Jerryescandon** [comment]
| Status | Copy on public | Colour |
| ------- | ---------------- | ----- |
| Draft  | "Closes in (time) days" | Gold | 
| Draft-expired  | "Closed on (date)" | Gold | 
| New application  | "Submitted and waiting for review"| Purple | 
| Application under review  | "Application under review" | Purple |
| Screened in (available)  | "Waiting for assessment" | Purple |
| Screened out - On application| "Application not accepted" | GRAY |
| Under assessment | "Waiting for assessment" | Purple |
| Screened out - On assessment  | "Assessment not passed" | GRAY |
| Qualified - Available  | "Qualified" | Navy |
| Qualified - Unavailable  | "Qualified" | Navy |
| Qualified - Withdrew  | "Qualified" | Navy |
| Placed Casual  | "Temporary placement" | Navy |
| Placed Term  | "Completed" | Navy |
| Placed Indeterminate  | "Completed" | Navy |
| Expired  | "Expired on (date)" | GRAY |
